### ISSUE TITLE
AdamW and SGD can work.

### ISSUE THREAD

**Issue by yuandong-tian** [Yuanforz/Grokking_reproduce#1]
Thanks for trying to reproducing my paper (https://arxiv.org/abs/2509.21519). I tested both AdamW and SGD and they work and can trigger grokking. 

Please use my repo (https://github.com/facebookresearch/luckmatters/) with yuandong3 branch. Use the following command line:

SGD:

```
python modular_addition_simple2.py activation=sqr loss_func=mse num_epochs=30000 weight_decay=2e-4 hidden_size=512 learning_rate=1 M=71 test_size=0.2 seed=1242 set_weight_reg=null save_interval=null optim=sgd
```

AdamW: 
```
python modular_addition_simple2.py activation=sqr loss_func=mse num_epochs=30000 weight_decay=2e-4 hidden_size=512 learning_rate=0.01 M=71 test_size=0.2 seed=1242 set_weight_reg=null save_interval=null optim=adamw
```

SGD needs larger learning rate.
